### DISCUSSION TITLE
How does the extension compatibility strategy look like?

### DISCUSSION THREAD

**Discussion by mygordienko** [eclipse-edc/Connector#736] [Q&A] [answered]
Looking at the _DemoS3FlowController_ I wonder how it can be sure that required fields are in the _DataAddress_ properties (ex. _bucketName_). I assume that providers and consumers can have different versions of the extensions. How does a consumer understand that it is supposed to send bucketName and object key separately and not in one URL?

**Comment by jimmarino** [comment, marked as answer]
Hi,

This will be part of the contract of a supported destination type. Destination addresses are similar to a URL where different schemes use different encoding mechanisms. For example, we will define what is required for an "S3" destination type, and the client will be expected to adhere to that.
